drivers/serial/uart_esp32.c

```diff
@@ -105,7 +105,7 @@ static int uart_esp32_poll_in(const struct device *dev, unsigned char *p_char)
 
 static void uart_esp32_poll_out(const struct device *dev, unsigned char c)
 {
-	const struct uart_esp32_config *config = dev->config;
+	struct uart_esp32_config *config = (struct uart_esp32_config *) dev->config;
```

Comment:
Why dropping `const`? Is driver using const data as non-const? If so, please open a bug.

Reply:
@gmarull All ESP32 `uarl_hal` calls discards cons qualifier. `uart_esp32_config` holds the `hal->dev` driver. That's the reason for removing the const in there. A second option would be moving `hal->dev` instance from `uart_esp32_config` to `uart_esp32_data`, then we keep this as const. Any thoughts?

Reply:
If `hal` contains data that is modified at runtime then yes, needs to be moved to `data`.

Reply:
@gmarull @glaubermaroto PTAL